network/quic/net.go

```diff
@@ -0,0 +1,160 @@
+package quic
+
+import (
+	"bufio"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"sync"
+	"time"
+
+	h "github.com/ConsenSys/handel"
+	"github.com/ConsenSys/handel/network"
+	quic "github.com/lucas-clemente/quic-go"
+)
+
+// Network is a handel.Network implementation using QUIC as its transport layer
+type Network struct {
+	mutex          sync.RWMutex
+	listeners      []h.Listener
+	quit           bool
+	enc            network.Encoding
+	quicListener   quic.Listener
+	sessionManager sessionManager
+}
+
+const handshakeTimeout = 2000 * time.Millisecond
+
+// NewNetwork creates Nework baked by QUIC protocol
+func NewNetwork(addr string, enc network.Encoding) (*Network, error) {
+	cfg := generateTLSConfig()
+	qCfg := &quic.Config{HandshakeTimeout: handshakeTimeout} //, AcceptCookie: f}
+	listener, err := quic.ListenAddr(addr, cfg, qCfg)
+
+	if err != nil {
+		panic(err)
+	}
+	var listeners []h.Listener
+	sessManager := newSessionManager(handshakeTimeout)
+	net := Network{
+		mutex:          sync.RWMutex{},
```

Comment:
It's just a tip but you don't have to initiliaze it, since that's the default value of the struct anyway. And you could also embed it in the struct as ```go
type Network struct { 
    sync.RWMutex
    ...
 }```
so it's directly callable as `network.Lock()`  
anyway, this is an informal comment xD